packages/jest-react/src/JestReact.js

```diff
@@ -35,19 +35,17 @@ function assertYieldsWereCleared(root) {
 }
 
 export function toFlushAndYield(root, expectedYields) {
+  assertYieldsWereCleared(root);
```

Comment:
I moved these outside because `captureAssertion` was swallowing the stack trace.